11_notify/willshen8/pkg/puppy/lostpuppy_test.go

```diff
@@ -0,0 +1,76 @@
+package puppy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func runTest(t *testing.T, payload []byte, expectedBody string, expectedHTTPCode int) {
+	req, err := http.NewRequest("POST", "/api/lostpuppy/", bytes.NewBuffer(payload))
+	if err != nil {
+		require.NoError(t, err)
```

Comment:
nit: There is no need to put the `require.NoError(t, err)` inside the `if err != nil` block. Just:
```
req, err := http.NewRequest(...)
require.NoError(t, err)
```